lib/platforms/platform.ts

```diff
@@ -0,0 +1,82 @@
+import { Message, MessageTypes } from './message';
+import { Logger } from './../logger';
+import diff from '../diff';
+import * as q from 'q';
+let logger = new Logger('Platform');
+export enum GroupTypes {
+    None,
+    PUBLIC,
+    PRIVATE,
+    PERSONAL
+}
+
+export class Platform {
+    protected normalizers = [];
+    protected messageTypes = {
+        FILE_SHARE: 'file_share'
+    };
+    public defaultRecipient: string;
+    public defaultMention: string;
+    constructor(...args) {
+        this.createNormalizers();
+    }
+
+    auth(): PromiseLike<Platform> {
+        return q.resolve(this);
```

Comment:
Why do we implement auth in Platform seems that this is always going to be platform dependent, including this noop which then gets used by slack seems like more of a hack rather than a feature. why not 
       logger.warn('Auth not implemented');
       return q.reject('auth: Method not implemented');